crates/core/src/dom/mod.rs

```diff
@@ -140,6 +159,17 @@ impl Document {
     pub fn edit(&mut self) -> Editor<'_> {
         Editor::new(self)
     }
+    pub fn merge_fragment(&mut self, root_diff: RootDiff) -> Result<Root, MergeError> {
+        let root = if let Some(root) = self.diff.as_mut() {
+            *root = root.clone().merge(root_diff)?;
+            root.clone()
```

Comment:
I'm wondering if the double clone is intentional here or could maybe be refactored?

Reply:
After describing this, I talked myself into `Result<(), MergeError>`.